### PR TITLE
Enforce ESLint (JavaScript linting) with pre-commit hooks

### PR DESCRIPTION
**Pull request by ryanahamilton**
The codebase has existing ESLint JavaScript linting rules configured, but they are not enforced (until now).

- I've added the `eslint` check to the pre-commit hooks.
- Refactored all of the existing files to be compliant with the rules so checks are passing.
- This is only being ran against external files (for now), as some additional refactoring will be required to get the inline scripting in order as well.
- I manually tested all of the webserver views that utilize these files and was unable to yield any console errors.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -75,7 +75,7 @@ <h2>DAGs</h2>
         <thead>
           <tr>
             <th width="12">
-              <span id="pause_header" class="material-icons text-muted" title="Use this toggle to pause a DAG. The scheduler won't schedule new tasks instances for a paused DAG. Tasks already running at pause time won't be affected.">info</span>
+              <span class="material-icons text-muted js-tooltip" title="Use this toggle to pause a DAG. The scheduler won't schedule new tasks instances for a paused DAG. Tasks already running at pause time won't be affected.">info</span>
```
Comment (ryanahamilton) [inline airflow/www/templates/airflow/dags.html]: This change is unrelated to this PR, but fixes a scripting interaction I discovered during my manual testing.

**Review by dimberman — APPROVED** [review]

**Comment by mik-laj** [comment]
Does this eslint configuration meet the project style guide?
https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#follow-javascript-style-guide

File: airflow/www/static/js/base.js [hunk 2]
```diff
@@ -47,57 +48,56 @@ function changDisplayedTimezone(tz) {
...
   return el.innerHTML;
 }
 
 window.escapeHtml = escapeHtml;
 
-export function convertSecsToHumanReadable(seconds) {
-  var oriSeconds = seconds;
-  var floatingPart = oriSeconds- Math.floor(oriSeconds);
+export function convertSecsToHumanReadable(secs) {
+  const oriSeconds = secs;
```
Comment (mik-laj) [inline airflow/www/static/js/base.js:60]: Do we have Babel? I know there were some compatibility issues with older browsers.

**Comment by ryanahamilton** [comment]
@mik-laj yes this is running the same configuration as `yarn run lint`, it just has a narrower scope (doesn't run against HTML files yet).

**Review by mik-laj — APPROVED** [review]

**Review by ashb — COMMENTED** [review]
```
/home/runner/work/airflow/airflow/airflow/www/static/js/ie.js
  22:8  error  Unable to resolve path to module 'url-search-params-polyfill'  import/no-unresolved

✖ 2 problems (1 error, 1 warning)


/home/runner/work/airflow/airflow/airflow/www/webpack.config.js
  20:25  error  Unable to resolve path to module 'webpack'                             import/no-unresolved
  22:32  error  Unable to resolve path to module 'webpack-manifest-plugin'             import/no-unresolved
  23:23  error  Unable to resolve path to module 'clean-webpack-plugin'                import/no-unresolved
  24:35  error  Unable to resolve path to module 'copy-webpack-plugin'                 import/no-unresolved
  25:38  error  Unable to resolve path to module 'mini-css-extract-plugin'             import/no-unresolved
  26:37  error  Unable to resolve path to module 'moment-locales-webpack-plugin'       import/no-unresolved
  27:41  error  Unable to resolve path to module 'optimize-css-assets-webpack-plugin'  import/no-unresolved
```

We might need to use something like https://www.npmjs.com/package/eslint-import-resolver-webpack ?

**Comment by ryanahamilton** [comment]
@ashb pre-commit wasn't including files outside of the `/js/` directory locally—I think I just need to add an explicit argument to restrict to that directory—though it might be preferred to do as you suggest and check/fix all of the files?

**Comment by potiuk** [comment]
> @ashb pre-commit wasn't including files outside of the `/js/` directory locally—I think I just need to add an explicit argument to restrict to that directory—though it might be preferred to do as you suggest and check/fix all of the files?

Yeah - we are running pre-commit with --all-files switch in CI.

**Comment by kaxil** [comment]
Rebased as Master was failing

**Comment by kaxil** [comment]
@ryanahamilton CI failed with the following:

```
ESLint............................................................................................................................Failed
- hook id: eslint
- exit code: 1

Error resolving webpackConfig Error: Cannot find module 'webpack'
Require stack:
- /home/runner/work/airflow/airflow/ai
.....
```

https://github.com/apache/airflow/pull/11699/checks?check_run_id=1292691649#step:7:176

**Comment by kaxil** [comment]
Can you please rebase your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

It will help if your squash your commits into single commit first so that there are less conflicts.

**Pull request closed without merging**